File: app/src/main/java/com/redhotapp/driverapp/data/remote/FcmService.java
package com.redhotapp.driverapp.data.remote;

import android.annotation.SuppressLint;
import android.util.Log;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.redhotapp.driverapp.data.local.preferences.Preferences;

import java.util.LinkedList;

@SuppressLint("NewApi")
public class FcmService extends FirebaseMessagingService {
  
  private static final String TAG = FcmService.class.getSimpleName();
  
  private static final int NOTIFICATION_ID = 1453;
  private static LinkedList<RemoteMessage> mNotifyData = new LinkedList<>();



  @Override
  public void onMessageReceived(RemoteMessage message) {

    Log.d(TAG, "++ FCM Message... latency (" + (System.currentTimeMillis() - message.getSentTime()) + " ms)");
    mNotifyData.add(message);
  }

  @Override
  public void onNewToken(String token) {
    Preferences.Companion.setFCMToken(getApplicationContext(), token);
    Log.i(TAG, "onNewToken()");
  }
}

File: app/src/main/java/com/redhotapp/driverapp/data/remote/FcmService.kt
package com.redhotapp.driverapp.data.remote

import android.annotation.SuppressLint
import android.util.Log
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.redhotapp.driverapp.data.local.preferences.Preferences
import java.util.*


@SuppressLint("NewApi")
class FcmService : FirebaseMessagingService() {
    override fun onMessageReceived(message: RemoteMessage) {
        Log.d(
            TAG,
            "++ FCM Message... latency (" + (System.currentTimeMillis() - message.sentTime) + " ms)"
        )
        mNotifyData.add(message)
    }

    override fun onNewToken(token: String) {
        Preferences.setFCMToken(
            applicationContext,
            token
        )
        Log.i(TAG, "onNewToken: " + token)
    }

    companion object {
        private val TAG = FcmService::class.java.simpleName
        private const val NOTIFICATION_ID = 1453
        private val mNotifyData =
            LinkedList<RemoteMessage>()
    }
}